src/main/java/seedu/address/logic/commands/ExpandPersonCommand.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Expands the selected person.
+ */
+public class ExpandPersonCommand extends Command {
+
+    public static final String COMMAND_WORD = "expandperson";
+
+    public static final String MESSAGE_SUCCESS = "Expanded: %1$s";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the person identified "
+            + "by the index number used in the displayed person list. \n"
+            + "Example: " + COMMAND_WORD + " 1 ";
+
+    private final Index index;
+
+    public ExpandPersonCommand(Index index) {
+        this.index = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToExpand = lastShownList.get(index.getZeroBased());
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
```

Comment:
I think this line should be removed since it negates search result. For instance, if I first search for all people with the surname 'Tan' with `findperson tan`, it gives me a list of everyone with the surname 'Tan'. Then if I choose to expand a person with `expandperson 1`, it expands the person, but the filtered list goes back to showing everyone in the address book. If the person is trying to review each person in a particular `find` query then it could be inconvenient for him. I'm not sure haha what do the rest of you guys think?

Reply:
Yup i agree